src/scan/lexer_public.ts

```diff
@@ -616,11 +619,14 @@ export class Lexer {
 
                 // mode handling, must pop before pushing if a Token both acts as both
                 // otherwise it would be a NO-OP
+                // need to save the PUSH_MODE property as if the mode is popped
+                // patternIdxToPopMode is updated to reflect the new mode after popping the stack
+                let pushMode = patternIdxToPushMode[i]
```

Comment:
the assignment here would happen for every token created.
even if for a large input there would be 50,000 tokens.

While this would have negligible effect on performance
It is probably better to avoid this and only perform the assignment in the case pop mode has happened.

I will refactor this in a separate change to do so.

This may seem like nitpicking but the lexer code is one of the most micro optimized code in chevrotain.